### PR TITLE
Caffeine Cache Manager - Expose cache names

### LINKED ISSUE #10612: [ Feature ] Return caches/cache names from CaffeineCacheManager
Current [CaffeineCacheManager](https://github.com/playframework/playframework/blob/master/cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala) implementation only exposes `getCache(name)` method, which is very limited for a cache manager.

adding a method that returns `cacheMap`, or returning cache names only, will make it much more useful since one can access raw Cache implementations & do anything I need to do, in case I need to. One common use case is to do same operation in all caches, like cleaning all caches (especially useful in tests ). Just because of this, we're currently using `EhCache` which exposes a `clearAll` method but cost 10Mb additional dependency ( since Play still uses v2.8 )

### PR DESCRIPTION
**Pull request by gokhanoner**
# Pull Request Checklist

* [x] Have you read [How to write the perfect pull request](https://github.com/blog/1943-how-to-write-the-perfect-pull-request)?
* [x] Have you read through the [contributor guidelines](https://www.playframework.com/contributing)?
* [x] Have you signed the [Lightbend CLA](https://www.lightbend.com/contribute/cla)?
* [x] Have you referenced any issues you're fixing using [commit message keywords](https://help.github.com/articles/closing-issues-using-keywords/)?
* [x] Have you added copyright headers to new files?
* [x] Have you checked that both Scala and Java APIs are updated?
* [x] Have you updated the documentation for both Scala and Java sections?
* [x] Have you added tests for any changed functionality?

# Helpful things

## Fixes

Fixes #10612 

## Purpose

Return names of available caches

## Background Context

Why did you take this approach?

## References

Are there any relevant issues / PRs / mailing lists discussions?
#10612

### CODE REVIEW

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by mkurz** [comment]
@gokhanoner The test you added isn't [properly formatted](https://travis-ci.com/github/playframework/playframework/jobs/471524266). In sbt, can you run `test:scalafmt` to fix that?

**Comment by gokhanoner** [comment]
@mkurz done.

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by mkurz** [comment]
@gokhanoner There is another problem: It seems you signed the CLA with the email address of your [GitHub account](https://github.com/gokhanoner)  which is ...@gmail.com but your commits where done with with a different email address (...@leantaas.com). That's why the typesafe-cla-validator fails. Either you add the other email address to your Github account and try to sign the CLA again or update the commits with the correct email address.

**Comment by lightbend-cla-validator** [comment]
At least one pull request committer is not linked to a user. See https://help.github.com/en/articles/why-are-my-commits-linked-to-the-wrong-user#commits-are-not-linked-to-any-user

**Comment by gokhanoner** [comment]
@mkurz I believe it's ready now.

File: cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala [hunk 1]
```diff
@@ -32,6 +34,10 @@ class CaffeineCacheManager(private val config: Config, private val actorSystem:
       .asInstanceOf[NamedCaffeineCache[K, V]]
   }
 
+  def getCacheNames(): Set[String] = {
+    Collections.unmodifiableSet(cacheMap.keySet());
+  }
```
Comment (mkurz) [inline cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala:39]: Actually I think this method should return a Scala `Set` and not a Java one.
/cc @octonato @ignasi35 WDYT?

Comment (octonato) [reply cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala:39]: Agree, this should return `scala.collection.immutable.Set`

Comment (octonato) [reply cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala:39]: An alternative is to follow the Akka project style. 

```scala
/* JAVA API */
def getCacheNames(): java.util.Set[String]

def cacheNames: scala.collection.immutable.Set[String]
```

Adding one method for the Java API and one for Scala. This is not the convention in Play, but there is no `CaffeineCacheManager` in the Java API.

Comment (gokhanoner) [reply cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala:39]: @mkurz @octonato I can change it to `scala.collection.immutable.Set`, no problem

Comment (mkurz) [reply cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala:36]: @gokhanoner I like @octonato's suggestion. Can you please change it once more so we end up with two methods, `getCacheNames()` returning the Java Set and `cacheNames` the Scala Set? Thanks. Afterwards I think its good to get merged :wink:

Comment (gokhanoner) [reply cache/play-caffeine-cache/src/main/scala/play/api/cache/caffeine/CaffeineCacheManager.scala:43]: @mkurz done.

**Review by mkurz — APPROVED** [review]

**Comment by mkurz** [comment]
@Mergifyio backport 2.8.x

**Pull request merged**